docs/buffer.md

```diff
@@ -20,29 +21,63 @@ Buffer is a [Node.js API](https://nodejs.org/dist/latest-v8.x/docs/api/buffer.ht
 to read and write binary data accurately from JavaScript. ZJS supports a minimal
 subset of this API that will be expanded as the need arises.
 
+Web IDL
+-------
+This IDL provides an overview of the interface; see below for documentation of
+specific API functions.  We have a short document explaining [ZJS WebIDL conventions](Notes_on_WebIDL.md).
+
+```javascript
+[ Constructor(Uint8Array initialValues),
+  Constructor(unsigned long size),
+  Constructor(ByteString initialString) ]
+interface Buffer {
+    readonly attribute unsigned long length;
+    unsigned long copy(Buffer target, optional unsigned long targetStart = 0,
+                                      optional unsigned long sourceStart = 0,
+                                      optional unsigned long sourceEnd);
+    this fill((string or Buffer or long) value, optional long offset = 0,
+                                                optional long end,
+                                                optional string encoding = "utf8");
+    octet readUInt8(optional unsigned long offset = 0);
```

Comment:
you cannot make anything optional and also give it a default value. It is one of the other